packages/styleguide/src/components/Preview/Interact.js

```diff
@@ -0,0 +1,838 @@
+              )
+            }
+          : {}),
+        ...Object.keys(liveProps).reduce((acc, curr) => {
+          const defaultValue =
+            docgenProps[curr].defaultValue &&
+            cleanValue(docgenProps[curr].defaultValue.value);
+          // default values should not be visible in code example
+          return {
+            ...acc,
+            [curr]:
+              defaultValue === liveProps[curr] || liveProps[curr] === ''
+                ? undefined
+                : liveProps[curr]
+          };
+        }, {})
+      }
+    };
+  }
+
+  /**
+   * Render function for Interact component
+   */
+  render() {
+    const componentIds = Object.keys(this.state.liveProps);
+
+    return (
+      <StyledInteract>
+        <Grid className="mb-large">
+          <GridCol
+            size={7}
+            className="align-items-middle align-items-middle"
+            style={{
+              borderRight: '1px solid #eaeaea',
+              position: 'relative'
+            }}
+          >
+            <StyledSticky>
+              <Bar>
+                <BarItem isFilling>
+                  <h3 className="h4 text-bold">
+                    {getDisplayName(this.component)}
+                  </h3>
+                </BarItem>
+                <BarItem>
+                  <Button
+                    onClick={this.handleShowCode}
+                    className={cx({ opened: this.state.showCode })}
+                  >
+                    {this.state.showCode ? 'Hide code ' : 'Show code '}
+                  </Button>
+                </BarItem>
+              </Bar>
+              {!this.state.showCode && this.renderInteractive(this.component)}
+              {this.state.showCode && (
+                <CodeExample
+                  codeJSXOptions={{
+                    cleanProps: true,
+                    filterProps: ['key']
+                  }}
+                >
+                  {this.renderInteractive(this.component)}
+                </CodeExample>
+              )}
+            </StyledSticky>
+          </GridCol>
+          <GridCol size={5}>
+            {componentIds.map((id, compIndex) => {
+              const statePropNames = Object.keys(this.docgen.liveProps[id]);
+              const propCount = statePropNames.length;
+              const deepness = id
+                .replace(/(\w+\d+)*(\w+)\d+$/g, '$1')
+                .replace(/\w+?\d+?/g, '-').length;
+              const componentName = id.replace(/(\w+\d+)*(\w+)\d+$/g, '$2');
+              return (
+                <div key={id} style={{ marginLeft: `${deepness * 10}px` }}>
+                  <Button
+                    onClick={e => {
+                      this.handleShowProps(e, id);
+                    }}
+                    className={cx({ opened: this.state.showProps[id] })}
+                  >
+                    {deepness > 0 && '↳'} {componentName} ({propCount}){' '}
```

Comment:
`({propCount})` ked som ho videl v prehliadaci, tak som nepochopil co to ciselko znaci. Myslel som si skor ze je to pokazene poradie, ked tam bolo viacero komponentov v preview :)

Reply:
vymazem to asi to tam netreba zatiaľ